src/driver/mysql/MysqlQueryRunner.ts

```diff
@@ -709,23 +709,37 @@ export class MysqlQueryRunner implements QueryRunner {
             case "decimal":
                 if (typeOptions.precision && typeOptions.scale) {
                     return `decimal(${typeOptions.precision},${typeOptions.scale})`;
-
                 } else if (typeOptions.scale) {
                     return `decimal(${typeOptions.scale})`;
-
                 } else if (typeOptions.precision) {
                     return `decimal(${typeOptions.precision})`;
-
                 } else {
                     return "decimal";
-
                 }
             case "date":
-                return "date";
+                if (!typeOptions.length) {
+                    return "date";
+                } else if (typeOptions.length <= 6) {
```

Comment:
It seems to me that we should not put this logic, because if user will set length to `10` but database will generate him `date(6)` he will be confused why its not generating him proper value, and he will not be aware of it. Probably better just to pass value back to db and left its validation on it. Alternatively we can make our own validation during schema create time.